src/main.rs

```diff
@@ -0,0 +1,206 @@
+        for r in result.iter() {
+            r.hash(&mut hasher);
+        }
+        let hash = hasher.finish();
+        if output.contains_key(&hash) {
+            return 0;
+        }
+
+        println!("{:?}", result);
+
+        output.insert(hasher.finish(), result);
+        return 1;
+    }
+
+    let mut rest = Vec::new();
+    rest.reserve(input.len() - 1);
+
+    let mut cur = prefix.to_vec();
+    cur.reserve(input.len());
+
+    let mut nb_results = 0;
+
+    let mut i = 0;
+    while i < input.len() {
+        rest.clear();
+        let mut j = 0;
+        while j < input.len() {
+            if j != i {
+                rest.push(input[j]);
+            }
+            j += 1;
+        }
+
+        cur.push(input[i]);
+        i += 1;
+
+        let key = last_word(&cur, spaces);
+        let mut prefixes = trie.iter_prefix(key.as_bytes()).take(1);
+        match prefixes.next() {
+            Some(_) => {}
+            None => {
+                cur.pop();
+                continue;
+            }
+        }
+
+        // Try also with a longer prefix.
+        nb_results += anagrams(input_size, &rest, &cur, spaces, trie, output);
+
+        if trie.contains(&key) {
+            // Current prefix is a known word. Add a space and continue.
+            let mut new_spaces = spaces.to_vec();
+            new_spaces.push(cur.len());
+            if spaces.len() < input_size / SPACES_FACTOR {
+                nb_results += anagrams(input_size, &rest, &cur, &new_spaces, trie, output);
+            }
+        }
+
+        cur.pop();
+    }
+
+    nb_results
+}
+
+fn main() {
+    let matches = App::new("Anagramme")
+        .author("Vincent C.")
+        .arg(
+            Arg::with_name("resource-dir")
+                .short("r")
+                .long("resource-dir")
+                .value_name("DIR")
+                .required(true)
+                .help("Path to the resource directory holding dictionary files")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("language")
+                .short("l")
+                .long("language")
+                .value_name("LANG")
+                .required(true)
+                .help("Language prefix (e.g. fr for french)")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("INPUT")
+                .help("Input sentence from which anagrams are tentatively found")
+                .required(true)
+                .index(1),
+        )
+        .get_matches();
+
+    let res_dir = matches.value_of("resource-dir").unwrap();
+    let input = matches.value_of("INPUT").unwrap().to_lowercase();
+    let lang = matches.value_of("language").unwrap().to_lowercase();
+
+    let txtfile = format!("{}.txt", lang);
+
+    let raw_dict_fr = get_raw_dict(txtfile, res_dir);
+    let mut trie = PatriciaSet::new();
+
+    let mut i = 0;
+    let mut start = 0;
+    while i < raw_dict_fr.len() {
```

Comment:
No need to declare or increment `i` yourself:
```suggestion
    for (i, item) in raw_dict_fr.iter().enumerate() {
```

Reply:
Ah! Thanks. I couldn't find the loop with increment and thought that Rust only add while loops while Go only has for loops :D